lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -106,6 +106,12 @@ class _Visitor extends SimpleAstVisitor<void> {
       // PrefixExpression.
       if (parent is MethodInvocation) {
         Expression target = parent.target;
+
+        // Something like `({1, 2, 3}).forEach(print);`.
+        // The parens cannot be removed because then the curly brackets are not
+        // interpreted as a set-or-map literal.
+        if (target == node && node.expression is SetOrMapLiteral) return;
```

Comment:
I think this disables the lint in too many places. Consider:
```
print({}.length);
```
No parentheses are needed here because it's not at the beginning of a statement. I think this condition should include ` && parent.parent is ExpressionStatement`.

Reply:
Thanks for this check, Brian! Your `.length` example also uncovered a PropertyAccess shortcoming in my impl. All fixed now.